nucypher/blockchain/eth/clients.py

```diff
@@ -186,37 +190,24 @@ def check_for_timeout(t):
                 raise self.SyncTimeout
 
         # Check for ethereum peers
-        self.log.info(f"Waiting for ethereum peers...")
+        self.log.info(f"Waiting for Ethereum peers ({len(self.peers)} known)")
         while not self.peers:
             time.sleep(0)
-            check_for_timeout(t=30)
-
-        needs_sync = False
-        for peer in self.peers:
-            peer_block_header = peer['protocols']['eth']['head']
-            try:
-                self.w3.eth.getBlock(peer_block_header)
-            except BlockNotFound:
-                needs_sync = True
-                break
-
-        # Start
-        if needs_sync:
-            peers = len(self.peers)
-            self.log.info(f"Waiting for sync to begin ({peers} ethereum peers)")
-            while not self.syncing:
-                time.sleep(0)
-                check_for_timeout(t=timeout)
-
-            # Continue until done
-            while self.syncing:
-                current = self.syncing['currentBlock']
-                total = self.syncing['highestBlock']
-                self.log.info(f"Syncing {current}/{total}")
-                time.sleep(1)
-                check_for_timeout(t=timeout)
-
-            return True
+            check_for_timeout(t=60)
+
+        # Wait for sync start
+        self.log.info(f"Waiting for {self.chain_name.capitalize()} chain synchronization to begin")
+        while not self.syncing:
+            time.sleep(0)
+            check_for_timeout(t=120)
+
+        while self.syncing:
+
+            # current =
+            self.log.info(f"Syncing {self.syncing['currentBlock']}/{self.syncing['highestBlock']}")
+            time.sleep(5)
```

Comment:
I have no concept of how long we expect this operation to be. If, as you said, it could be in such a wide range, seconds -> hours, then perhaps sleeping for 5 or 10s is sufficient. Just wasn't sure if there would be cases of us waiting for a long window when it took somewhere in the order of ms.